### PR TITLE
[Feature][flink-connector] support flink  delete option

### PR DESCRIPTION
**Pull request by JNSimba**
## Proposed changes

Support Flink delete options

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/extending-doris/flink-doris-connector.md [hunk 1]
```diff
@@ -300,6 +300,7 @@ outputFormat.close();
 | sink.max-retries     | 1                | 写BE失败之后的重试次数       |
 | sink.batch.interval     | 1s                | flush 间隔时间，超过该时间后异步线程将 缓存中数据写入BE。 默认值为1秒，支持时间单位ms、s、min、h和d。设置为0表示关闭定期写入。|
 | sink.properties.*     | --               | Stream load 的导入参数<br /><br />例如:<br />'sink.properties.column_separator' = ', '<br />定义列分隔符<br /><br />'sink.properties.escape_delimiters' = 'true'<br />特殊字符作为分隔符,'\\x01'会被转换为二进制的0x01<br /><br /> 'sink.properties.format' = 'json'<br />'sink.properties.strip_outer_array' = 'true' <br />JSON格式导入|
+| sink.enable-delete     | false               | 是否启用删除。此选项需要Doris表开启批量删除功能|
```
Comment (aihai) [inline docs/zh-CN/extending-doris/flink-doris-connector.md:303]: How to set the delete flag, if using the datastream api.

File: extension/flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicOutputFormat.java [hunk 2]
```diff
@@ -113,24 +133,26 @@ public DorisDynamicOutputFormat(DorisOptions option,
...
         }
 
-        this.fieldNames = fieldNames;
-        this.jsonFormat = FORMAT_JSON_VALUE.equals(executionOptions.getStreamLoadProp().getProperty(FORMAT_KEY));
-        this.fieldGetters = new RowData.FieldGetter[logicalTypes.length];
-        for (int i = 0; i < logicalTypes.length; i++) {
-            fieldGetters[i] = createFieldGetter(logicalTypes[i], i);
+        //add column key
+        if (!streamLoadProp.containsKey(COLUMNS_KEY)) {
+            String columns = String.join(",", Arrays.stream(fieldNames).map(item -> String.format("`%s`", item.trim().replace("`", ""))).collect(Collectors.toList()));
```
Comment (aihai) [inline extension/flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicOutputFormat.java:138]: if using datastream api, fieldNames should be defined.

Comment (JNSimba) [reply extension/flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicOutputFormat.java:138]: Thank you for reminding me that I ignored the datastream submission method

**Comment by aihai** [comment]
I am worry about whether the key word `__DORIS_DELETE_SIGN__` is compatible in the later version.

**Comment by morningman** [comment]
> I am worry about whether the key word `__DORIS_DELETE_SIGN__` is compatible in the later version.

It will not change.

**Comment by morningman** [comment]
hi @JNSimba , Could you please rebase the master and resolve the conflict?

**Comment by JNSimba** [comment]
> hi @JNSimba , Could you please rebase the master and resolve the conflict?

thanks, resolved

**Review by hf200012 — COMMENTED** [review]
Give examples of delete usage

**Comment by JNSimba** [comment]
> Give examples of delete usage

Thanks  remind，already added

**Review by hf200012 — DISMISSED** [review]
LGTM

**Comment by JNSimba** [comment]
I will make a modification later: only when it is judged to be a unique model, perform the delete operation.

**Comment by JNSimba** [comment]
only when it is judged to be a unique model, perform the delete operation.

**Review by hf200012 — DISMISSED** [review]
LGTM

**Comment by JNSimba** [comment]
Compatible with history Doris versions

**Review by hf200012 — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**